### DISCUSSION TITLE
I would like to program X1nput to modulate rumble signals

### DISCUSSION THREAD

**Discussion by LeVraiRoiDHyrule** [araghon007/X1nput#20] [Ideas]
Hi,

I just tried X1nput with some of my games, it works great !

But I would like to custom it a bit more. My objective would be to feel rumbling waves on my controller instead of just impulsion on the 4 motors. Currently, when a game sends a rumble signal, it justs turn all the motors on and off at the same time. I would like to modulate the rumble signals.

I would like to use sin() and cos() functions to modulate the signals, so the rumbling is oscillating. I would also love to be able to delay the signals on each motors. That way, I would be able to program rumble from left to right, or right to left. That would make the rumbling feedback much more immersive.

But I don't know where I could do this. Does someone have an idea for such a project ? Thanks in advance for your help

**Comment by araghon007** [comment]
Hello,
I tried making a UE4 plugin at one point, but quickly found out that's not a good idea. I wouldn't recommend using the approach I used here for actual game, since it's only functional on windows 10, and requires the game window to be in focus. I tried experimenting with the approach people used for the new DualSense triggers, but gave up on that one since I have no experience with USB protocols.

**Comment by LeVraiRoiDHyrule** [comment]
Well, your plugin works great. I would just like to know if it's possible to use like motor1 = sin(t+3.14)*output instead of motor1 = output. Maybe it can be simply edited in the plugin ? Maybe you already tried...

**Comment by araghon007** [comment]
If you're experienced with C++, you could try integrating this into your project. You'll have to do the initialization stuff from dllmain, and then look at XInputSetState to see how to set vibration values on the controller. Otherwise I don't really know how to help you.

**Comment by araghon007** [comment]
I just realized that that's sort of what lindquest did. I'm hoping I could experiment with custom curves at some point, but I'm not much of a mathematician. Also let me see what does this button do
